src/devices/Vcnl4040/ProximitySensor.cs

```diff
@@ -0,0 +1,346 @@
+            _psConf3Register = new PsConf3Register(device);
+            _psMsRegister = new PsMsRegister(device);
+            _psCancellationLevelRegister = new PsCancellationLevelRegister(device);
+            _psLowInterruptThresholdRegister = new PsLowInterruptThresholdRegister(device);
+            _psHighInterruptThresholdRegister = new PsHighInterruptThresholdRegister(device);
+            _psDataRegister = new PsDataRegister(device);
+            _psWhiteDataRegister = new WhiteDataRegister(device);
+            _alsConfRegister = new AlsConfRegister(device);
+        }
+
+        #region General
+
+        /// <summary>
+        /// Get or sets the power state (power on, shutdown) of the proximity sensor.
+        /// </summary>
+        public bool PowerOn
+        {
+            get
+            {
+                _psConf1Register.Read();
+                return _psConf1Register.PsSd == PowerState.PowerOn;
+            }
+
+            set
+            {
+                _psConf1Register.Read();
+                if ((value && _psConf1Register.PsSd == PowerState.PowerOn)
+                || (!value && _psConf1Register.PsSd == PowerState.PowerOff))
+                {
+                    return;
+                }
+
+                _psConf1Register.PsSd = value ? PowerState.PowerOn : PowerState.PowerOff;
+                _psConf1Register.Write();
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the state of the active force mode for proximity measurement.
+        /// When the sensor is operated in Active Force Mode, an explicit request for a measurement
+        /// is required before a current measurement value can be read.
+        /// This is in contrast to normal operation, where measurements are continuous.
+        /// This significantly reduces power consumption, as the IR LED is only activated for each
+        /// individual measurement.
+        /// Important: In Active Force Mode, the interrupt function is not available,
+        /// as no measurement is performed, and therefore, threshold checks are not possible.
+        /// </summary>
+        public bool ActiveForceMode
+        {
+            get
+            {
+                _psConf3Register.Read();
+                _activeForceModeEnabled = _psConf3Register.PsAf == PsActiveForceMode.Enabled;
+                return _activeForceModeEnabled;
+            }
+
+            set
+            {
+                _psConf3Register.PsAf = value ? PsActiveForceMode.Enabled : PsActiveForceMode.Disabled;
+                _psConf3Register.Write();
+                _activeForceModeEnabled = value;
+            }
+        }
+        #endregion
+
+        #region Measurement
+
+        /// <summary>
+        /// Gets the current proximity sensor reading.
+        /// Note: if active force mode is enabled reading this property implicitly triggers
+        /// the measurement for once cycle.
+        /// </summary>
+        public int Reading
```

Comment:
This isn't more expensive than any other property that has access through I2C. The reading is always available, even if the actual measurement process an the device takes a significant time. So it would stick to the property. But, I agree, that "Distance" or "Proximity" would be a more precise name for it.